server/src/main/java/com/linecorp/centraldogma/server/internal/storage/repository/git/GitRepository.java

```diff
@@ -1590,6 +1599,44 @@ private void readUnlock() {
         rwLock.readLock().unlock();
     }
 
+    @VisibleForTesting
+    boolean writeLock(boolean directExecution) {
+        // Should not commit changes if gc is running
+        boolean hasGcLock;
+        if (directExecution) {
+            try {
+                // Waits only 10 seconds the gc lock to prevent a command worker queue from filing up
+                // due to a long gc, if the command was directly submitted to this server.
+                hasGcLock = gcLock.tryLock() || gcLock.tryLock(10, TimeUnit.SECONDS);
```

Comment:
`gcLock.tryLock()` does not honor the fairness.
Shouldn't we honor the fairness because there can be multiple commits waiting?

Reply:
I don't think of that case. It would be better to remove `tryLock()` for that.